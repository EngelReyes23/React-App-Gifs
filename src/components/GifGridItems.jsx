import React from "react";
import PropTypes from "prop-types";

const GifGridItems = ({ title, url }) => {
	return (
		<div className="card animate__animated animate__fadeIn">
			<a href={url} target="_blank" rel="noreferrer">
				<img src={url} alt={title} />
			</a>
			<p>{title}</p>
		</div>
	);
};

GifGridItems.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
};

export default GifGridItems;
