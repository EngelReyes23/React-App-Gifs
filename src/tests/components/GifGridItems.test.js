import { shallow } from "enzyme";
import GifGridItems from "../../components/GifGridItems";

describe("Pruebas en <GifGrindItems />", () => {
	//
	const wrapper = shallow(<GifGridItems title="titulo" url="url" />);

	//
	test("Debe de mostrar el componente correctamente", () => {
		expect(wrapper).toMatchSnapshot();
	});

	//
	test("debe tener un parrafo con el title", () => {
		const p = wrapper.find("p");
		expect(p.text().trim()).toBe("titulo");
	});

	//
	test("debe tener un src con el url y un alt con el titulo", () => {
		const url = wrapper.find("img").props();
		expect(url.src).toBe("url");
		expect(url.alt).toBe("titulo");
	});

	//
	test("debe tener la clase animate__fadeIn", () => {
		const className = wrapper.find("div").prop("className");
		expect(className.includes("animate__fadeIn")).toBe(true);
	});
});
