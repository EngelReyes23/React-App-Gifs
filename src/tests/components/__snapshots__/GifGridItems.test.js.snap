// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrindItems /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <a
    href="url"
    rel="noreferrer"
    target="_blank"
  >
    <img
      alt="titulo"
      src="url"
    />
  </a>
  <p>
    titulo
  </p>
</div>
`;
